### PR TITLE
Release 2.4.0

### PR DESCRIPTION
**Pull request by Daniel-VM**
## PR checklist

- [X] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/bacass/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/bacass _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

## PR description

Release 2.4.0

Major changes are:
- **Refactored long-reads polishing** – Improved the long-reads polishing step for better performance.
- **Removed `params.save_merged`** – Merged reads are no longer used in downstream analysis.
- **Updated Kmerfinder module** – Enhanced the module with increased memory resources to handle larger datasets.
- **Replaced local Unicycler module** – Switched to the nf-core Unicycler module for consistency and version updates.
- **Updated to nf-core-tools v3.0.2** – Integrated the latest template updates from nf-core tools v3.0.2.
- **Added FastQC after-trimming** – Included after-trimming FastQC section in MultiQC report for improved quality checks.
- **Supported automatic concatenation of FastQ files** – Simplified input handling by adding automatic concatenation for FastQ files of the same sample.
- **Multiple fixes in input and metadata handling** – Resolved issues in input channels for long reads, `params.save_merged`, Canu, KmerFinder, and Prokka/Bakta. 

This summary covers the main improvements, new features, and fixes in version 2.4.0 of the `nf-core/bacass` pipeline.

### CODE REVIEW

File: workflows/bacass.nf [hunk 1]
```diff
@@ -71,11 +45,24 @@ include { PROKKA                                } from '../modules/nf-core/prokk
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    CONFIG FILES
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+// Check input path parameters to see if they exist
+def checkPathParamList = [ params.input, params.multiqc_config, params.kraken2db, params.dfast_config ]
+for (param in checkPathParamList) { if (param) { file(param, checkIfExists: true) } }
+
+// Place config files here
```
Comment (d4straub) [inline workflows/bacass.nf:63]: is that a reminder? Is that solved?

**Review by d4straub — COMMENTED** [review]
Looks good, but I do have comments ;)

File: .github/workflows/awsfulltest.yml [hunk 2]
```diff
@@ -1,18 +1,35 @@
...
 jobs:
   run-platform:
     name: Run AWS full tests
-    if: github.repository == 'nf-core/bacass'
+    # run only if the PR is approved by at least 2 reviewers and against the master branch or manually triggered
+    if: github.repository == 'nf-core/bacass' && github.event.review.state == 'approved' && github.event.pull_request.base.ref == 'master' || github.event_name == 'workflow_dispatch'
     runs-on: ubuntu-latest
     steps:
+      - uses: octokit/request-action@v2.x
+        id: check_approvals
```
[the remark this answers is not in the document]
Comment (d4straub) [reply .github/workflows/awsfulltest.yml:22]: Sure, add to new PRs, whatever works/is convenient

File: workflows/bacass.nf (hunk 1 above)
Comment (Daniel-VM) [reply workflows/bacass.nf:63]: Solved ;). This was a leftover from an nf-core-tools update.

**Comment by Daniel-VM** [comment]
Thanks for the review. @d4straub, suggestions have been added here: https://github.com/nf-core/bacass/pull/184

**Review by d4straub — APPROVED** [review]
LGTM!

**Pull request merged**